docs/Cluster-Internal-networks.md

```diff
@@ -0,0 +1,42 @@
+# Cluster Internal Network design
+
+This document contains the information used to decide whether to provision separate (unique) networks for each OpenShift 4 cluster.  The current design requires routeable (public) IP addresses only at the F5 ingress/egress points for each cluster, and we are not looking to change the design of using non-routeable IP addresses for the following networks.  The decision in question is whether we can re-use the same private network ranges for multiple clusters.  This decision supports a critical cluster component, which if changed, would require a full re-installation of a given cluster.
+
+## Background
+
+With each OpenShift cluster, internal networks are used for in-cluster communications.  These networks are private networks that rely on ingress and egress Network Address Translation.  If a network does not require external direct routing/exposure, then we can save on IP network reservations by re-using these private networks.  The following networks are the potential private networks for OCP 4:
+
+- Host/Node Network - primary physical interface for each physical host in a cluster (outgoing NAT IP address for all pods on a node)
+- Storage Network - connecting NAS storage (NetApp) to each of the hosts within a cluster.
+- Pod Network - IP addresses assigned to each pod.
+- Service Network - IP addresses assigned to services.
+
+## Network usage
+
+### Host/Node Network
+
+Host IP addresses are used to NAT each hosted PODs outgoing traffic, as well as host specific node/cluster related services.  Incoming application traffic will not be directed at a host IP address.  While we are planning to use outgoing NAT on the F5, there are situations where we may want to be able to direct route to these host IP addresses from inside the data-centres.  Current OCP3 leverages public IP addresses for all host/nodes so a pre-build list of services that would require direct access has not been described/tested yet.
+
+A separate (unique) Host/Node network for each cluster is recommended.
```

Comment:
A separate unique private routable Host/Node network for each cluster is recommended.